### ISSUE TITLE
[VersionControl] Crash adding files to the project

### ISSUE THREAD

**Issue by jsuarezruiz** [mono/monodevelop#7101]
Adding a big ammount of files, and/or a big file and trying to access to the Version Control contextual menu.

```
at <unknown> <0xffffffff>  at LibGit2Sharp.Core.NativeMethods:git_stash_save <0x00144>  
at LibGit2Sharp.Core.Proxy:git_stash_save <0x000ca>  at LibGit2Sharp.StashCollection:Add <0x00122>  
at MonoDevelop.VersionControl.Git.GitRepository:TryCreateStash <0x000f7>  
at MonoDevelop.VersionControl.Git.GitRepository:CommonPreMergeRebase <0x00442>  
at MonoDevelop.VersionControl.Git.GitRepository:Rebase <0x00102>  
at MonoDevelop.VersionControl.Git.GitRepository:OnUpdate <0x00192>  
at MonoDevelop.VersionControl.Repository:Update <0x0015c>  
at UpdateWorker:Run <0x000d2>  at MonoDevelop.VersionControl.VersionControlTask:<BackgroundWorker>b__13_0 <0x00010>  
at System.Threading.Tasks.Task:InnerInvoke <0x00054>  at System.Threading.Tasks.Task:Execute <0x00023>  
at System.Threading.Tasks.Task:ExecutionContextCallback <0x00052>  
at System.Threading.ExecutionContext:RunInternal <0x0020d>  
at System.Threading.ExecutionContext:Run <0x0002a>  
at System.Threading.Tasks.Task:ExecuteWithThreadLocal <0x00122>  
at System.Threading.Tasks.Task:ExecuteEntry <0x000e2>  
at System.Threading.Tasks.Task:System.Threading.IThreadPoolWorkItem.ExecuteWorkItem <0x0001a>  
at System.Threading.ThreadPoolWorkQueue:Dispatch <0x002e9>  
at System.Threading._ThreadPoolWaitCallback:PerformWaitCallback <0x00052>  
at <Module>:runtime_invoke_bool <0x00075>
```

Regression #7078

> VS bug [#789298](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/789298)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.1594 (master)

Pull Request #7078 merged by: Vsevolod Kukol
Author: mono
Commit: e9006c97e60bc51b1fe82dec9f80c9eb4f57b7cc (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.2351 (release-8.0)

Pull Request #7078 merged by: Vsevolod Kukol
Author: mono
Commit: e9006c97e60bc51b1fe82dec9f80c9eb4f57b7cc (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.2351 (release-8.0)

Pull Request #7078 merged by: Vsevolod Kukol
Author: mono
Commit: e9006c97e60bc51b1fe82dec9f80c9eb4f57b7cc (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #7078 merged by: Vsevolod Kukol
Author: mono
Commit: e9006c97e60bc51b1fe82dec9f80c9eb4f57b7cc (xamarin/monodevelop)
